mesonbuild/compilers/cuda.py

```diff
@@ -326,13 +732,24 @@ def get_crt_link_args(self, crt_val: str, buildtype: str) -> T.List[str]:
         host_crt_compile_args = self.host_compiler.get_crt_compile_args(crt_val, buildtype)
         if any(arg in ['/MDd', '/MD', '/MTd'] for arg in host_crt_compile_args):
             host_link_arg_overrides += ['/NODEFAULTLIB:LIBCMT.lib']
-        return self._cook_link_args(host_link_arg_overrides + self.host_compiler.get_crt_link_args(crt_val, buildtype))
+        return self._to_host_flags(host_link_arg_overrides + self.host_compiler.get_crt_link_args(crt_val, buildtype), _Phase.LINKER)
 
     def get_target_link_args(self, target: 'BuildTarget') -> T.List[str]:
-        return self._cook_link_args(super().get_target_link_args(target))
+        return self._to_host_flags(super().get_target_link_args(target), _Phase.LINKER)
 
     def get_dependency_compile_args(self, dep: 'Dependency') -> T.List[str]:
         return self._to_host_flags(super().get_dependency_compile_args(dep))
 
     def get_dependency_link_args(self, dep: 'Dependency') -> T.List[str]:
-        return self._cook_link_args(super().get_dependency_link_args(dep))
+        return self._to_host_flags(super().get_dependency_link_args(dep), _Phase.LINKER)
+
+    def get_ccbin_args(self, options: 'KeyedOptionDictType') -> T.List[str]:
+        key = OptionKey('ccbindir', machine=self.for_machine, lang=self.language)
+        try:
+            ccbindir = options[key].value
+            if isinstance(ccbindir, str) and ccbindir != '':
+                return [self._shield_nvcc_list_arg('-ccbin='+ccbindir, False)]
+            else:
+                return []
```

Comment:
I'd pull the `if/else` out of the try/except block, since now if there were an unhandled `KeyError `in `self._shield_nvcc_list_arg`, it would get silently caught here.

Reply:
I had this `try` only because I had an inexplicable `KeyError` with key `ccbindir` somewhere in the Arch CUDA CI, despite the flag being defaulted to empty and thus "absent", not changing the CI to try to use it, and the log messages all saying the CUDA tests had _passed_.

I'll try to strike it out.

Reply:
@dcbaker Could you please have a look at that failure on the Arch CUDA runner and tell me what I'm doing wrong?

I introduced a new `-Dcuda_ccbindir=path` option, see `mesonbuild/compilers/cuda.py:582`, and it works for me in practice, but `run_unittests.py` chokes on it. I suspect it's bad mocking, but did I do everything I was supposed to do in order to register this new option?

Reply:
That took me a while to figure out, the problem is that unittest is skipping some important stuff to get the compiler, replace the `getattr(env, 'detect_foo_comiler')` line with `env.detect_compiler_for(l, MachineChoice.HOST)`, that should fix it.

Reply:
@dcbaker You were right; The Arch CUDA runner has just passed.